modules/c++/mem/include/mem/VectorOfPointers.h

```diff
@@ -116,10 +117,24 @@ class VectorOfPointers
     iterator end() { return mValues.end(); }
     const_iterator end() const { return mValues.end(); }
 
+    iterator erase(const iterator &pos)
+    {
+        delete *pos;
+        return mValues.erase(pos);
+    }
+
+    iterator erase(const iterator &first, const iterator &last)
+    {
+        std::for_each(first, last, Deleter());
```

Comment:
So this is a style/opinion thing... If you could use a lambda here it would save you a few lines and `for_each()` would buy you something.  With something this simple, granted the name `Deleter` is obvious, but I have to go look at the struct to make sure and I think it's simpler code to skim over by just using a for loop.  Style-wise, we'll want curly braces even for one liner for loops FYI.